File: intersect.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
  let nums = [],
    min = nums1,
    max = nums2
  if(nums1.length > nums2.length) {
    min = nums2
    max = nums1
  }
  min.forEach(num => {
    let index = max.indexOf(num)
    if (index > -1) {
      nums.push(num)
      max.splice(index, 1)
    }
  })
  return nums
}

File: intersect.go
package main

func intersect(nums1 []int, nums2 []int) []int {
	var nums []int
	numsMap := make(map[int]int, len(nums1))
	for _, v := range nums1 {
		if _, ok := numsMap[v]; ok {
			numsMap[v]++
		} else {
			numsMap[v] = 1
		}
	}
	for _, t := range nums2 {
		if _, ok := numsMap[t]; ok {
			if numsMap[t] > 0 {
				nums = append(nums, t)
				numsMap[t]--
			}
		}
	}
	return nums
}

func main() {
}
